src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.IO;
+using Newtonsoft.Json;
+using Microsoft.AspNetCore.Http;
+using System.Net.Http.Headers;
+
+namespace Microsoft.R.Host.Broker.RemoteUri {
+    public class RemoteUriHelper {
+        public static async Task HandlerAsync(HttpContext context) {
+            string url = null;
+            string method = null;
+            Dictionary<string, string> headers = null;
+            Stream content = null;
+
+            BinaryReader reader = new BinaryReader(context.Request.Body);
+            url = reader.ReadString();
+            method = reader.ReadString();
+            headers = JsonConvert.DeserializeObject<Dictionary<string, string>>(reader.ReadString());
+
+            HttpClient client = new HttpClient();
+            Uri uri = new Uri(url);
+
+            HttpMethod httpMethod = new HttpMethod(method);
+            HttpRequestMessage req = new HttpRequestMessage(httpMethod, uri);
+
+            foreach (var key in headers.Keys) {
+                var val = headers[key];
+                req.Headers.Add(key, val);
+            }
+
+            if (httpMethod == HttpMethod.Post) {
```

Comment:
Is there any particular reason to restrict this to Post? It should definitely also handle at least Put, but I think it could be method-blind and just forward the body as is - it should be empty for other methods anyway (and if not, it's not really this code's problem).